plugins/kernel_v2/config/vm.rb

```diff
@@ -218,7 +218,7 @@ def synced_folder(hostpath, guestpath, options=nil)
           synced_folder_name = guestpath
         end
 
-        options[:guestpath] = guestpath.to_s.gsub(/\/$/, '')
+        options[:guestpath] = guestpath.to_s.gsub(/\/$/, '') if guestpath
```

Comment:
This should be a config validation error, rather than the option being silently ignored if the guestpath isn't specified (or is empty) with a synced_folders config option.

Reply:
It should not be a config validation error. It is supposed to be an optional parameter but is not currently treated as such. Please see https://github.com/hashicorp/vagrant/pull/7980 as well as my issue write-up. Thanks!

Reply:
Hi @cyungmann - yes, sorry for taking a while to get back to you. This is a bit tricky, because the `name` option isn't documented at all so it isn't clear how this should really work, and what might break with this change. I plan on adding some documentation around the `name` option soon (likely with this change).

I think you're change is good, so that we don't attempt to create empty string folders...but I still believe there should be a validation error thrown for a potential invalid configuration state to warn users before vagrant fails out and shows ugly stack traces. I understand `guesthpath` is supposed to be optional, but if it isn't provided then it seems like `name` is required? Otherwise the option breaks in fun ways when it assumes it has a `guestpath` or `name` to mount a share with. It would also be good to get some tests in this class with this change to ensure that if name or guestpath is not provided it is an invalid configuration as well as test that if `name` is provided, the guestpath option is `nil` and not empty string.

If you need any help with that let me know, thanks!

Reply:
Hello, @briancain. Thank you for the feedback! It all made sense to me, and you are right that things blow up if neither `name` nor `guestpath` are provided. I updated the PR with your suggested changes. Please let me know if I should change anything (I'm a total Ruby newbie!).